### ISSUE TITLE
SIGSEV on pipelinerun logs 

### ISSUE THREAD

**Issue by afrittoli** [tektoncd/cli#775]
# Version and Operating System

**tkn Version:**
```
Client version: 0.8.0
Pipeline version: v0.10.1
```
**Operating System:** 
MacOS Catalina

# Expected Behavior
I can view logs of the pipeline

# Actual Behavior
```
$ tkn pr logs tekton-noop-check-fcxr6 -n tektonci
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1d68c24]

goroutine 1 [running]:
github.com/tektoncd/cli/pkg/helper/log.(*Reader).readAvailablePipelineLogs(0xc0000cac80, 0xc00000ad20, 0x17, 0x0, 0x0, 0x0)
	/workspace/src/github.com/tektoncd/cli/pkg/helper/log/pipeline_reader.go:90 +0xc4
github.com/tektoncd/cli/pkg/helper/log.(*Reader).readPipelineLog(0xc0000cac80, 0x8, 0x21cfec8, 0x7ffeefbff51c, 0xc0003c38f0)
	/workspace/src/github.com/tektoncd/cli/pkg/helper/log/pipeline_reader.go:41 +0x13d
github.com/tektoncd/cli/pkg/helper/log.(*Reader).Read(0xc0000cac80, 0x8, 0xc0001b92c0, 0xc0000cac80, 0x0)
	/workspace/src/github.com/tektoncd/cli/pkg/helper/log/reader.go:77 +0xb8
github.com/tektoncd/cli/pkg/cmd/pipelinerun.Run(0xc0001b92c0, 0xc0002ee000, 0x0)
	/workspace/src/github.com/tektoncd/cli/pkg/cmd/pipelinerun/logs.go:105 +0x81
github.com/tektoncd/cli/pkg/cmd/pipelinerun.logCommand.func1(0xc00030f8c0, 0xc000324e70, 0x1, 0x3, 0x0, 0x0)
	/workspace/src/github.com/tektoncd/cli/pkg/cmd/pipelinerun/logs.go:78 +0x1ba
github.com/spf13/cobra.(*Command).execute(0xc00030f8c0, 0xc000324de0, 0x3, 0x3, 0xc00030f8c0, 0xc000324de0)
	/workspace/pkg/mod/github.com/chmouel/cobra@v0.0.0-20200107083527-379e7a80af0c/command.go:834 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc00029c840, 0xc0002ee000, 0xc00029c840, 0xc00004c118)
	/workspace/pkg/mod/github.com/chmouel/cobra@v0.0.0-20200107083527-379e7a80af0c/command.go:922 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/workspace/pkg/mod/github.com/chmouel/cobra@v0.0.0-20200107083527-379e7a80af0c/command.go:872
main.main()
	/workspace/src/github.com/tektoncd/cli/cmd/tkn/main.go:29 +0x4b
```

# Additional Info

I suspect this is because the pipeline definition is embedded with `pipelineSpec`

Spec of the pipelinerun:
```
apiVersion: tekton.dev/v1alpha1
kind: PipelineRun
metadata:
  creationTimestamp: "2020-03-04T14:57:11Z"
  generation: 1
  labels:
    prow.k8s.io/build-id: 6c7c2700-5e28-11ea-9bdf-639ed7fddffb
    tekton.dev/check-name: tekton-noop-check
    tekton.dev/eventlistener: tekton-ci-webhook
    tekton.dev/kind: ci
    tekton.dev/pipeline: tekton-noop-check-fcxr6
    tekton.dev/pr-number: "254"
    tekton.dev/trigger: pull-request-trigger
    tekton.dev/triggers-eventid: 8mnlv
  name: tekton-noop-check-fcxr6
  namespace: tektonci
  resourceVersion: "57546407"
  selfLink: /apis/tekton.dev/v1alpha1/namespaces/tektonci/pipelineruns/tekton-noop-check-fcxr6
  uid: c13de874-f41e-40ac-92b0-42cab6b17a6a
spec:
  pipelineSpec:
    resources:
    - name: source
      type: git
    - name: starttrigger
      type: cloudEvent
    - name: endtrigger
      type: cloudEvent
    tasks:
    - name: check-reset
      params:
      - name: checkName
        value: tekton-noop-check
      resources:
        outputs:
        - name: trigger
          resource: starttrigger
      taskRef:
        name: tekton-ci-reset-check-status
    - name: ci-job
      params:
      - name: message
        value: 'BuildUUID#: 6c7c2700-5e28-11ea-9bdf-639ed7fddffb, Package: tektoncd/plumbing,
          Repo: $(params.gitRepository, Revision: 24e4ebb02519a8433a23551c9cc4fb4af8a8d1eb'
      resources:
        inputs:
        - name: source
          resource: source
        outputs:
        - name: endtrigger
          resource: endtrigger
      taskRef:
        name: tekton-noop
  resources:
  - name: source
    resourceSpec:
      params:
      - name: revision
        value: 24e4ebb02519a8433a23551c9cc4fb4af8a8d1eb
      - name: url
        value: https://https://github.com/tektoncd/plumbing
      type: git
  - name: starttrigger
    resourceRef:
      name: github-check-start
  - name: endtrigger
    resourceRef:
      name: github-check-done
  serviceAccountName: tekton-ci-jobs
  timeout: 1h0m0s
status:
  completionTime: "2020-03-04T14:57:19Z"
  conditions:
  - lastTransitionTime: "2020-03-04T14:57:17Z"
    message: TaskRun tekton-noop-check-fcxr6-ci-job-jrwtz has failed
    reason: Failed
    status: "False"
    type: Succeeded
  startTime: "2020-03-04T14:57:11Z"
  taskRuns:
    tekton-noop-check-fcxr6-check-reset-mwv7n:
      pipelineTaskName: check-reset
      status:
        cloudEvents:
        - status:
            condition: Sent
            message: ""
            retryCount: 1
            sentAt: "2020-03-04T14:57:19Z"
          target: http://el-github-check-start.tektonci:8080
        completionTime: "2020-03-04T14:57:19Z"
        conditions:
        - lastTransitionTime: "2020-03-04T14:57:19Z"
          message: All Steps have completed executing
          reason: Succeeded
          status: "True"
          type: Succeeded
        podName: tekton-noop-check-fcxr6-check-reset-mwv7n-pod-28wps
        startTime: "2020-03-04T14:57:11Z"
        steps:
        - container: step-noop
          imageID: docker-pullable://busybox@sha256:edafc0a0fb057813850d1ba44014914ca02d671ae247107ca70c94db686e7de6
          name: noop
          terminated:
            containerID: docker://9b1704d9a47ee8d5db0a5c2dc9caefefb75195057a6a6c6c3174dc3504a8e689
            exitCode: 0
            finishedAt: "2020-03-04T14:57:18Z"
            reason: Completed
            startedAt: "2020-03-04T14:57:18Z"
        - container: step-create-dir-trigger-6r4pp
          imageID: docker-pullable://busybox@sha256:edafc0a0fb057813850d1ba44014914ca02d671ae247107ca70c94db686e7de6
          name: create-dir-trigger-6r4pp
          terminated:
            containerID: docker://9e7cdc2935f059ccd52deb8f7b0f0d758d8af9d7a5e9f1a4b5d76f2aed4c1534
            exitCode: 0
            finishedAt: "2020-03-04T14:57:17Z"
            reason: Completed
            startedAt: "2020-03-04T14:57:17Z"
    tekton-noop-check-fcxr6-ci-job-jrwtz:
      pipelineTaskName: ci-job
      status:
        cloudEvents:
        - status:
            condition: Sent
            message: ""
            retryCount: 1
            sentAt: "2020-03-04T14:57:17Z"
          target: http://el-github-check-done.tektonci:8080
        completionTime: "2020-03-04T14:57:17Z"
        conditions:
        - lastTransitionTime: "2020-03-04T14:57:17Z"
          message: '"step-git-source-source-kvqqp" exited with code 1 (image: "docker-pullable://gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init@sha256:18ffa2bfc14b1fa6d39f62271beacf6bbc38e7cd2e255184dec477c2936270bc");
            for logs run: kubectl -n tektonci logs tekton-noop-check-fcxr6-ci-job-jrwtz-pod-tm2rb
            -c step-git-source-source-kvqqp'
          reason: Failed
          status: "False"
          type: Succeeded
        podName: tekton-noop-check-fcxr6-ci-job-jrwtz-pod-tm2rb
        startTime: "2020-03-04T14:57:11Z"
        steps:
        - container: step-noop
          imageID: docker-pullable://busybox@sha256:edafc0a0fb057813850d1ba44014914ca02d671ae247107ca70c94db686e7de6
          name: noop
          terminated:
            containerID: docker://c6343b395ac30d72be1aa10ed67e83f251523e334e650f2af955bbc98d8feb49
            exitCode: 1
            finishedAt: "2020-03-04T14:57:16Z"
            reason: Error
            startedAt: "2020-03-04T14:57:15Z"
        - container: step-git-source-source-kvqqp
          imageID: docker-pullable://gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init@sha256:18ffa2bfc14b1fa6d39f62271beacf6bbc38e7cd2e255184dec477c2936270bc
          name: git-source-source-kvqqp
          terminated:
            containerID: docker://420d6648b5a80d0f4e6a0bfc7d70a2d34c62ddaf37f25a9a0cc1971ebbaf4095
            exitCode: 1
            finishedAt: "2020-03-04T14:57:16Z"
            reason: Error
            startedAt: "2020-03-04T14:57:16Z"
        - container: step-create-dir-endtrigger-4x7nm
          imageID: docker-pullable://busybox@sha256:edafc0a0fb057813850d1ba44014914ca02d671ae247107ca70c94db686e7de6
          name: create-dir-endtrigger-4x7nm
          terminated:
            containerID: docker://d898391ac0a26ce97e2dc75084cfd2d65be38f44e2bdba169f89d30c9479fcc1
            exitCode: 0
            finishedAt: "2020-03-04T14:57:16Z"
            reason: Completed
            startedAt: "2020-03-04T14:57:16Z"
```

**Comment by 16yuki0702** [comment]
> I suspect this is because the pipeline definition is embedded with pipelineSpec

yes, I can reprduce this bug using embedded pipelineSpec.
I'll fix this.

**Comment by 16yuki0702** [comment]
maybe this is same https://github.com/tektoncd/cli/issues/603

**Comment by siamaksade** [comment]
I ran into the same issue with 0.8.0 but master seem to have resolved it already:

0.8.0:
```
$ tkn pr describe condtional-pr
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1d33a86]

goroutine 1 [running]:
github.com/tektoncd/cli/pkg/helper/pipelinerun/description.taskrunList.Less(0xc0004268d0, 0x2, 0x2, 0x1, 0x0, 0xc0003d3a10)
	/workspace/src/github.com/tektoncd/cli/pkg/helper/pipelinerun/description/description.go:85 +0x66
sort.insertionSort(0x235e680, 0xc0003fb220, 0x0, 0x2)
	/usr/local/go/src/sort/sort.go:27 +0xc4
sort.quickSort(0x235e680, 0xc0003fb220, 0x0, 0x2, 0x4)
	/usr/local/go/src/sort/sort.go:209 +0x201
sort.Sort(0x235e680, 0xc0003fb220)
	/usr/local/go/src/sort/sort.go:218 +0x79
github.com/tektoncd/cli/pkg/helper/pipelinerun/description.PrintPipelineRunDescription(0xc0003d3d40, 0x7ffeefbff980, 0xd, 0x23761a0, 0xc0002dc100, 0x0, 0x0)
	/workspace/src/github.com/tektoncd/cli/pkg/helper/pipelinerun/description/description.go:126 +0x883
github.com/tektoncd/cli/pkg/cmd/pipelinerun.describeCommand.func1(0xc000300dc0, 0xc0003106c0, 0x1, 0x1, 0x0, 0x0)
	/workspace/src/github.com/tektoncd/cli/pkg/cmd/pipelinerun/describe.go:73 +0x20b
github.com/spf13/cobra.(*Command).execute(0xc000300dc0, 0xc000310690, 0x1, 0x1, 0xc000300dc0, 0xc000310690)
	/workspace/pkg/mod/github.com/chmouel/cobra@v0.0.0-20200107083527-379e7a80af0c/command.go:834 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002d62c0, 0xc0002dc100, 0xc0002d62c0, 0xc0000460b8)
	/workspace/pkg/mod/github.com/chmouel/cobra@v0.0.0-20200107083527-379e7a80af0c/command.go:922 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/workspace/pkg/mod/github.com/chmouel/cobra@v0.0.0-20200107083527-379e7a80af0c/command.go:872
main.main()
	/workspace/src/github.com/tektoncd/cli/cmd/tkn/main.go:29 +0x4b
```


master:
```
# ./bin/tkn pr describe condtional-pr
Name:              condtional-pr
Namespace:         demo
Pipeline Ref:      conditional-pipeline
Service Account:   default
Timeout:           1h0m0s

🌡️  Status

STARTED        DURATION   STATUS
1 minute ago   1 minute   Succeeded

📦 Resources

 NAME            RESOURCE REF
 ∙ source-repo   pipeline-git

⚓ Params

 No params

🗂  Taskruns

 NAME                                      TASK NAME           STARTED          DURATION     STATUS
 ∙ condtional-pr-then-check-fxcrr          then-check          52 seconds ago   14 seconds   Succeeded
 ∙ condtional-pr-first-create-file-2qw6m   first-create-file   1 minute ago     41 seconds   Succeeded
```

**Comment by 16yuki0702** [comment]
> I ran into the same issue with 0.8.0 but master seem to have resolved it already:

maybe you mean about ```pr describe``` ?
(and this PR is related ? https://github.com/tektoncd/cli/pull/780)

this issue about ```pr logs```.

**Comment by danielhelfand** [comment]
The `tkn pr describe` issue is unrelated to #780. It seems similar to what was reported in #584 and resolved in #586. 

If that's occurring in `v0.8.0` and can be reproduced, then that should be looked into further as #586 was part of v0.7.1.

**Comment by danielhelfand** [comment]
Closing this from #781
